### PR TITLE
Extauth gme parity

### PR DESCRIPTION
**Pull request by sheidkamp**
# Description
Extauth Internal API changes to support parity with GMG

## API changes
* Internal API Changes:
  * Move `Azure` message up a level in scope to be directly under the top level `ExtAuthConfig` for reuse across internal config types
  * Move `ClaimToHeader` message up a level in scope to be directly under the top level `ExtAuthConfig` for reuse across internal config types
  * `AccessTokenValidationConfig` message
    * Add dynamic_metadata_from_claims
      * Supports [dynamicMetadataFromClaims in oauth2 accesstokenvalidation](https://github.com/solo-io/gloo/issues/9282)
  * `OidcAuthorizationCodeConfig` 
    * Add dynamic_metadata_from_claims
      * Supports [dynamicMetadataFromClaims in oauth2](https://github.com/solo-io/gloo/issues/9282)
    * Add `claimsToHeaders` field
      * Supports [Distributed claims for OIDC Access Code authentication](https://github.com/solo-io/gloo-mesh-enterprise/issues/18590)
    * Add  `provider` to support OIDC distributed claims
      *  Supports [Distributed claims for OIDC Access Code authentication](https://github.com/solo-io/gloo-mesh-enterprise/issues/18590) 
  * Add ServerDefaultApiKeyStorage to internal ApiKeyAuthConfig storage_backends.
    * Supports use of[ Redis backend/server default storage for ApiKeys
](https://github.com/solo-io/solo-projects/issues/5451)

## Testing steps
In [solo-project PR](https://github.com/solo-io/solo-projects/pull/7091) (to be added): 

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto [hunk 1]
```diff
@@ -1080,6 +1080,10 @@ message AccessTokenValidation {
         // in the future we may add other types of scope validation (e.g. predicate matching)
     }
 
+    message ScopeList {
+      repeated string scope = 1;
+    }
+
```
Comment (sheidkamp) [inline projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto:1086]: This was moved to keep structure consistent, no other changes.

**Comment by sheidkamp** [comment]
Moving to wip while Access Token validation OIDC code changes are added

**Review by sam-heilbron — COMMENTED** [review]
LGTM

**Review by sam-heilbron — APPROVED** [review]

**Comment by sheidkamp** [comment]
/kick https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=ee278dc8-bdbc-4bce-b23e-6fa3a3524cd8

```
Step #0 - "publish-artifacts": Step #4 - "publish-docker": Get "https://quay.io/v2/": net/http: request canceled (Client.Timeout exceeded while awaiting headers)
Step #0 - "publish-artifacts": Step #4 - "publish-docker": make: *** [Makefile:957: docker-push-sds] Error 1
```

**Review by conradhanson — APPROVED** [review]

**Pull request merged**